src/wagtail_vector_index/storage/base.py

```diff
@@ -152,6 +164,55 @@ def query(
         response = chat_backend.chat(messages=messages)
         return QueryResponse(response=response.choices[0], sources=sources)
 
+    async def aquery(
```

Comment:
mostly a copy of the existing `query` method, but uses async calls to the underlying LLM API and to the database